### ISSUE TITLE
Does the author have right to publish the book again after accepting pull requests?

### ISSUE THREAD

**Issue by CyrusYip** [rstudio/rmarkdown-book#72]
This a nice book! I noticed that it is an opensource book licensed under 
Attribution-NonCommercial-ShareAlike 4.0 International. It means others can send pull requests to improve it, and the author( @yihui ) can earn money from it. I think it is a good practice of earning money from writing opensource books. I also want to write such a book.

It is no doubt that the author can commercially sell books under Attribution-NonCommercial-ShareAlike 4.0 if he or she writes the wholebook. 

Say that I wrote an opensource book under Attribution-NonCommercial-ShareAlike 4.0 . I wonder whether I can sell my book after accepting others' pull requests? Can I use the pull request content commercially?

**Comment by yihui** [comment]
First things first: I'm not a lawyer.

Most of the time the pull requests are quite simple, and the most typical ones are those fixing typos. I don't think you'd have problems with these. For pull requests that are substantial, you can ask their authors to provide an agreement (CLA) to transfer the copyright to you. This can be handled automatically by https://cla-assistant.io.

**Comment by CyrusYip** [comment]
Thank you, @yihui. Your advice is helpful. I found another way: claiming that **contributions are donations in the contribution guidelines**. 

[exerpt of The JavaScript Way Contribution guilines:](https://github.com/thejsway/thejsway/blob/master/CONTRIBUTING.md)

> A merged PR will make you appear in the contributor list. It will, however, be considered a donation of your work to this project. You are still bound by the conditions of the license, meaning that you are not considered an author or owner of the content once it has been merged in.

There is a relevant question on StackExchange:  [What is the appropriate license for open-source content which I might publish into a book?](https://opensource.stackexchange.com/questions/9082/what-is-the-appropriate-license-for-open-source-content-which-i-might-publish-in/10522#10522). I wrote my understanding of the question there.
